sacred/utils.py

```diff
@@ -315,6 +316,58 @@ def __init__(
         super().__init__(message, print_traceback, filter_traceback, print_usage)
 
 
+class FilteredTracebackException(tb.TracebackException):
+    def __init__(
+        self,
+        exc_type,
+        exc_value,
+        exc_traceback,
+        *,
+        limit=None,
+        lookup_lines=True,
+        capture_locals=False,
+        _seen=None,
+    ):
+        exc_traceback = self._filter_tb(exc_traceback)
+        if exc_value.__cause__:
+            exc_value.__cause__.__traceback__ = self._filter_tb(
+                exc_value.__cause__.__traceback__
+            )
+        if exc_value.__context__:
+            exc_value.__context__.__traceback__ = self._filter_tb(
+                exc_value.__context__.__traceback__
+            )
+
+        super().__init__(
+            exc_type,
+            exc_value,
+            exc_traceback,
+            limit=limit,
+            lookup_lines=lookup_lines,
+            capture_locals=capture_locals,
+            _seen=_seen,
+        )
+
+    def _filter_tb(self, tb):
+        filtered_tb = []
+        while tb is not None:
+            if not _is_sacred_frame(tb.tb_frame):
+                filtered_tb.append(
+                    TracebackType(None, tb.tb_frame, tb.tb_lasti, tb.tb_lineno)
```

Comment:
It looks like Python 3.6 doesn't allow creating instances of `traceback` or `TracebackType`...

Reply:
Yes, that came with 3.7. I also can't create a subclass and the `__traceback__` needs to be of type `TracebackType`. I don't see a way around it.

How about dropping support for Python 3.6, EOL is at the end of the year anyway?